src/Workspaces/Core/Portable/FindSymbols/SymbolFinder.cs

```diff
@@ -15,7 +15,11 @@ namespace Microsoft.CodeAnalysis.FindSymbols
     public static partial class SymbolFinder
     {
         /// <summary>
-        /// Finds the symbol that is associated with a position in the text of a document.
+        /// Finds the symbol that is associated with a position in the text of a document. If 
+        /// multiple symbols are associated with the token (e.g. implicitly named anonymous type
```

Comment:
And David's change illustrates why "list of symbols" is an icky API: GetSymbols today returns (I _think_):

1) the declaration you're touching
2) the symbol you're referencing, if different
3) the alias of the referenced symbol, if that exists
4) the type symbol of this, if you happened to be touching 'this'
5) something about delegates I don't even get

My preference for API design here is we give the simple API for the simple case, and good building blocks for the special cases. We already have APIs on SemanticModel to ask for definition vs. reference vs. alias, we should just let consumers use those if they need it. I think the this -> type of this was added for Quick Info, when it's not clear to me if that policy should apply in general.